### PR TITLE
feat(iam): improvements to esxi password management

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

New customer focused functions to cover expiration, complexity and account lockout as three distinct configuration options for ESXi.
- Request-EsxiPasswordExpiration
- Request-EsxiPasswordComplexity
- Request-EsxiAccountLockout
- Update-EsxiPasswordExpiration
- Update-EsxiPasswordComplexity
- Update-EsxiAccountLockout

- Created a new region for Password Management Functions for easier management

Moved the following functions to the deprecated section:
- Get-EsxiPasswordPolicy
- Set-EsxiPasswordExpirationPeriod
- Set-EsxiPasswordPolicy

Signed-off-by: Gary Blake <gblake@vmware.com>

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Release History
 
+## v2.0.0 (Not Released)
+
+- Added `Request-EsxiPasswordExpiration` cmdlet to retrieve the password expiration policy for all ESXi hosts in a cluster.
+- Added `Request-EsxiPasswordComplexity` cmdlet to retrieve the password complexity policy for all ESXi hosts in a cluster.
+- Added `Request-EsxiAccountLockout` cmdlet to retrieve the account lockout policy for all ESXi hosts in a cluster.
+- Added `Update-EsxiPasswordExpiration` cmdlet to update the password expiration policy for all ESXi hosts in a cluster.
+- Added `Update-EsxiPasswordComplexity` cmdlet to update the password complexity policy for all ESXi hosts in a cluster.
+- Added `Update-EsxiAccountLockout` cmdlet to update the account lockout policy for all ESXi hosts in a cluster.
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply CHANGELOG.md:10]: For the deprecation yes we should add to the change log if we are going to pull them at some point, right now I've placed a load of functions in a deprecation section but none have been removed as such yet. We could add the list for this release and then remove in 2.1.0

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -16086,6 +15861,524 @@ Export-ModuleMember -Function Update-SddcDeployedFlavor
...
+											} else {
+												Write-Error "Unable to retrieve password expiration policy from ESXi host ($esxiHost.Name): PRE_VALIDATION_FAILED"
+											}
+										}
+										return $esxiPasswdPolicy
+									} else {
+										Write-Warning "No ESXi hosts found within cluster named ($cluster): PRE_VALIDATION_FAILED"
+									}
+								} else {
+                                    Write-Error "Unable to locate Cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:15929]:
```suggestion
                                    Write-Error "Unable to locate cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -16086,6 +15861,524 @@ Export-ModuleMember -Function Update-SddcDeployedFlavor
...
+											} else {
+												Write-Error "Unable to retrieve password complexity policy from ESXi host ($esxiHost.Name): PRE_VALIDATION_FAILED"
+											}
+										}
+										return $esxiPasswdPolicy
+									} else {
+										Write-Warning "No ESXi hosts found within cluster named ($cluster): PRE_VALIDATION_FAILED"
+									}
+								} else {
+                                    Write-Error "Unable to locate Cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:16010]:
```suggestion
                                    Write-Error "Unable to locate cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -16086,6 +15861,524 @@ Export-ModuleMember -Function Update-SddcDeployedFlavor
...
+											} else {
+												Write-Error "Unable to retrieve account lockout policy from ESXi host ($esxiHost.Name): PRE_VALIDATION_FAILED"
+											}
+										}
+										return $esxiPasswdPolicy
+									} else {
+										Write-Warning "No ESXi hosts found within cluster named ($cluster): PRE_VALIDATION_FAILED"
+									}
+								} else {
+                                    Write-Error "Unable to locate Cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:16091]:
```suggestion
                                    Write-Error "Unable to locate cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)): PRE_VALIDATION_FAILED"
```

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -16086,6 +15861,524 @@ Export-ModuleMember -Function Update-SddcDeployedFlavor
...
+                                            if ($detail -eq "true") {
+                                                Write-Warning "Update Password Complexity Policy (Security.PasswordHistory) to ($history) on ESXi Host ($esxiHost), already set: SKIPPED"
+                                            }
+                                        }
+                                    }
+                                    if ($detail -eq "false") {
+                                        Write-Output "Update Password Complexity Policy (Security.PasswordQualityControl and Security.PasswordHistory) on all ESXi Hosts for Workload Domain ($domain): SUCCESSFUL"
+                                    }
+                                } else {
+                                    Write-Error "Unable to find Cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)), check details and retry: PRE_VALIDATION_FOUND"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:16265]:
```suggestion
                                    Write-Error "Unable to find cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)), check details and retry: PRE_VALIDATION_FOUND"
```

File: PowerValidatedSolutions.psm1 [hunk 6]
```diff
@@ -16086,6 +15861,524 @@ Export-ModuleMember -Function Update-SddcDeployedFlavor
...
+                                            if ($detail -eq "true") {
+                                                Write-Warning "Update Password Complexity Policy (Security.AccountUnlockTime) to ($unlockTime) on ESXi Host ($esxiHost), already set: SKIPPED"
+                                            }
+                                        }
+                                    }
+                                    if ($detail -eq "false") {
+                                        Write-Output "Update Password Complexity Policy (Security.AccountLockFailures and Security.AccountUnlockTime) on all ESXi Hosts for Workload Domain ($domain): SUCCESSFUL"
+                                    }
+                                } else {
+                                    Write-Error "Unable to find Cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)), check details and retry: PRE_VALIDATION_FOUND"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:16359]:
```suggestion
                                    Write-Error "Unable to find cluster ($cluster) in vCenter Server ($($vcfVcenterDetails.fqdn)), check details and retry: PRE_VALIDATION_FOUND"
```

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply CHANGELOG.md:10]: That sounds excellent to me.

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply CHANGELOG.md:10]: I'll mark this as resolved and we can add the changes to the CHANGELOG in a separate pull request.

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀 

I only added a couple case suggestions, but those are not important.

**Pull request merged**